### PR TITLE
=str #20009 State shared over multiple materializations eliminated

### PR DESCRIPTION
**Pull request by johanandren**
Fixes the two internal bugs not fixed yet in #20009

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala [hunk 1]
```diff
@@ -194,6 +194,27 @@ class QueueSourceSpec extends AkkaSpec {
       queue.offer(1).onFailure { case e ⇒ e.isInstanceOf[IllegalStateException] should ===(true) }
     }
 
+    "not share future across materializations" in {
+      val source = Source.queue[String](1, OverflowStrategy.fail)
+
+      val mat1subscriber = TestSubscriber.probe[String]()
+      val mat2subscriber = TestSubscriber.probe[String]()
+      val sourceQueue1 = source.to(Sink.fromSubscriber(mat1subscriber)).run()
+      val sourceQueue2 = source.to(Sink.fromSubscriber(mat2subscriber)).run()
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala]: would `TestSink.probe` be simpler here?

**Comment by ktoso** [comment]
LGTM, thanks for noticing those

**Comment by rkuhn** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM

**Comment by johanandren** [comment]
MiMa fail, moved promise field was public, whitelisted.

**Comment by rkuhn** [comment]
Hmm, perhaps implement it with an UnsupportedOperationException that explains that previous usage was broken?

**Comment by rkuhn** [comment]
or was it marked as INTERNAL API?

**Comment by johanandren** [comment]
The entire `QueueSource` class is.

**Pull request merged**